compiler/lib/php_from_rehp.re

```diff
@@ -301,16 +316,29 @@ let rec foldSources = (sourceFolder, curOut, curIn, curRevMappeds, remain) =>
   };
 
 let rec foldStatements =
-        (statementFolder, curOut, curIn, curRevMappeds, remain) =>
+        (
+          statementFolder,
+          origOut: output,
+          curOut: output,
+          curIn: input,
+          curRevMappeds,
+          remain,
+        ) =>
   switch (remain) {
   | [] => (curOut, List.rev(curRevMappeds))
   | [(s, loc), ...tl] =>
-    let (thisOut, thisMapped) = statementFolder(curOut, curIn, s);
+    let (thisOut, thisMapped) =
+      statementFolder(
+        {...curOut, unshielded_continues: origOut.unshielded_continues},
+        curIn,
+        s,
+      );
     let nextOut = outAppend(curOut, thisOut);
-    let nextIn = append(curIn, nextOut.dec);
+    let nextIn = {vars: append(curIn.vars, nextOut.dec), label: curIn.label};
```

Comment:
If I have a set of statements, I don't think that labels in one statement are in the "scope" of subsequent labels are they?

```
a:
for(;;) {

};

continue a;
```
In that example, when we're folding over the set of statements, `a` isn't in the scope of the second statement right? This is different from variable scoping, which is why we previously had ` let nextIn = append(curIn, nextOut.dec);`